### PR TITLE
HBASE-25496 add get_namespace_rsgroup command

### PR DESCRIPTION
**Pull request by leyangyueshan**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -0 :warning: |  rubocop  |   0m 12s |  The patch generated 9 new + 370 unchanged - 0 fixed = 379 total (was 370)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |   3m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | dupname asflicense rubocop |
| uname | Linux bddbc09b1942 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| rubocop | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 46 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/console |
| versions | git=2.17.1 maven=3.6.3 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 47s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 33s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m  1s |  hbase-shell in the patch passed.  |
|  |   |  17m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux dd8dc85b972c 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/testReport/ |
| Max. process+thread count | 2337 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 18s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  8s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 52s |  hbase-shell in the patch passed.  |
|  |   |  18m 16s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux 1d517733fb95 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/testReport/ |
| Max. process+thread count | 2411 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Example:
+
+  hbase> get_namespace_rsgroup 'namespace_name'
+
+EOF
+      end
+
+      def command(namespace_name)
+        group_name = admin.get_namespace_rsgroup(namespace_name)
+        formatter.row(group_name)
```
Comment (virajjasani) [inline hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:34]: Do we need nil check here?

File: hbase-shell/src/main/ruby/hbase/admin.rb [hunk 2]
```diff
@@ -1447,6 +1447,17 @@ def alter_namespace(namespace_name, *args)
...
 
+    #----------------------------------------------------------------------------------------------
+    # Get namespace's rsgroup
+    def get_namespace_rsgroup(namespace_name)
+      # Fail if namespace name is not a string
+      raise(ArgumentError, 'Namespace name must be of type String') unless namespace_name.is_a?(String)
+      nsd = @admin.getNamespaceDescriptor(namespace_name)
+      raise(ArgumentError, 'Namespace does not exist') unless nsd
+      res = nsd.getConfigurationValue("hbase.rsgroup.name")
+      res
```
Comment (virajjasani) [inline hbase-shell/src/main/ruby/hbase/admin.rb:1458]: Can be replaced with single line:
```
nsd.getConfigurationValue("hbase.rsgroup.name")
```

Comment (leyangyueshan) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1458]: no, I tried, but it doesn't work。
like rsgroup_admin.rb 
  #--------------------------------------------------------------------------
    # get group of server
    def get_rsgroup_of_server(server)
      res = @admin.getRSGroup(
        org.apache.hadoop.hbase.net.Address.fromString(server)
      )
      raise(ArgumentError, 'Server has no group: ' + server) if res.nil?
      res
    end

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb (hunk 1 above)
Comment (leyangyueshan) [reply hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:34]: thanks ,  no need, I think , pr has undated!

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -0 :warning: |  rubocop  |   0m 13s |  The patch generated 10 new + 370 unchanged - 0 fixed = 380 total (was 370)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  8s |  The patch does not generate ASF License warnings.  |
|  |   |   2m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | dupname asflicense rubocop |
| uname | Linux d3d0571e1315 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| rubocop | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 42 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/console |
| versions | git=2.17.1 maven=3.6.3 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shell/src/main/ruby/hbase/admin.rb (hunk 2 above)
Comment (virajjasani) [reply hbase-shell/src/main/ruby/hbase/admin.rb:1458]: I meant this:
```
    def get_namespace_rsgroup(namespace_name)
      # Fail if namespace name is not a string
      raise(ArgumentError, 'Namespace name must be of type String') unless namespace_name.is_a?(String)
      nsd = @admin.getNamespaceDescriptor(namespace_name)
      raise(ArgumentError, 'Namespace does not exist') unless nsd
      nsd.getConfigurationValue("hbase.rsgroup.name")
    end
```

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 30s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m  2s |  hbase-shell in the patch passed.  |
|  |   |  16m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux 0b12ace335bf 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/testReport/ |
| Max. process+thread count | 2328 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 20s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  6s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 57s |  hbase-shell in the patch passed.  |
|  |   |  17m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux b3344e360161 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/testReport/ |
| Max. process+thread count | 2209 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Example:
+
+  hbase> get_namespace_rsgroup 'namespace_name'
+
+EOF
+      end
+
+      def command(namespace_name)
+        group_name = admin.get_namespace_rsgroup(namespace_name)
```
Comment (virajjasani) [inline hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:33]: How will `group_name` be printed? I think you just removed print statement.

File: hbase-shell/src/main/ruby/hbase/admin.rb (hunk 2 above)
Comment (leyangyueshan) [reply hbase-shell/src/main/ruby/hbase/admin.rb]: yes, pr has updated， please review again！

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -0 :warning: |  rubocop  |   0m 13s |  The patch generated 9 new + 370 unchanged - 0 fixed = 379 total (was 370)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 13s |  The patch does not generate ASF License warnings.  |
|  |   |   2m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | dupname asflicense rubocop |
| uname | Linux eb508c4f4984 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| rubocop | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/console |
| versions | git=2.17.1 maven=3.6.3 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb (hunk 3 above)
Comment (leyangyueshan) [reply hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:33]: image

I code it from get_server_rsgroup method, and it look good !

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 49s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 29s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 59s |  hbase-shell in the patch passed.  |
|  |   |  16m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux 27458f42c140 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/testReport/ |
| Max. process+thread count | 2374 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 34s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 17s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 15s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   6m 55s |  hbase-shell in the patch passed.  |
|  |   |  17m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux 15d33ea1c710 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 686b72c44e |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/testReport/ |
| Max. process+thread count | 2364 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb (hunk 3 above)
Comment (virajjasani) [reply hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:33]: With same changes, can you try searching for namespace which doesn't have any rsgroup?
e.g if namespace 'NMn' exists but it is not part of any rsgroup (rsgroup feature is not enabled on cluster), then what will `get_namespace_rsgroup 'NMn'` return?

**Review by virajjasani — COMMENTED** [review]
Left one comment. Otherwise changes look good.

File: hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb (hunk 3 above)
Comment (leyangyueshan) [reply hbase-shell/src/main/ruby/shell/commands/get_namespace_rsgroup.rb:33]: thanks a lot， I will update this pr

**Comment by leyangyueshan** [comment]
> Left one comment. Otherwise changes look good.

pr has updated ! add nil check, thank you very much !
I checked

**Comment by leyangyueshan** [comment]
> Left one comment. Otherwise changes look good.

pr has updated ! add nil check ! I check two situtation as you said
return nothing
image

thank you very much ! @virajjasani

**Comment by virajjasani** [comment]
Great, taking a look.

**Review by virajjasani — APPROVED** [review]
+1

**Comment by virajjasani** [comment]
Let's wait for some time as few releases are going on.
Thanks for the changes

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 43s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 44s |  the patch passed  |
| -0 :warning: |  rubocop  |   0m 15s |  The patch generated 10 new + 370 unchanged - 0 fixed = 380 total (was 370)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  14m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | dupname asflicense javac rubocop |
| uname | Linux d58463ebc9bd 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / dfefff7e59 |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
| rubocop | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/artifact/yetus-general-check/output/diff-patch-rubocop.txt |
| Max. process+thread count | 65 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/console |
| versions | git=2.17.1 maven=3.6.3 rubocop=0.80.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 19s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 27s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m  0s |  hbase-shell in the patch passed.  |
|  |   |  16m 58s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux 9ca54cf3a871 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / dfefff7e59 |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/testReport/ |
| Max. process+thread count | 2349 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  9s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 32s |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 18s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m  0s |  hbase-shell in the patch passed.  |
|  |   |  20m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2874 |
| Optional Tests | javac javadoc unit |
| uname | Linux df269cc7a5f7 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / dfefff7e59 |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/testReport/ |
| Max. process+thread count | 2366 (vs. ulimit of 30000) |
| modules | C: hbase-shell U: hbase-shell |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2874/4/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**